### DISCUSSION TITLE
My own proxy station deployed through Docker still request sum.golang.org when in use

### DISCUSSION THREAD

**Discussion by Tryanks** [goproxy/goproxy#53] [Q&A] [answered]
My proxy is deployed on an accelerated server and can be accessed via https://go.XXXXX.com
Locally set:
```
go env -w GOPROXY=https://go.XXXXX.com,direct
```
When synchronizing projects with `go mod tidy`, each synchronization still takes a long time and fails after a period of time. Here is an excerpt from the logs:
```
(base) ~\GolandProjects\project git:[finish]
go mod tidy
go: downloading github.com/stretchr/testify v1.3.0
go: downloading github.com/google/pprof v0.0.0-20221118152302-e6195bd50e26
go: downloading modernc.org/ccgo/v4 v4.10.1
go: downloading modernc.org/fileutil v1.3.0
go: downloading github.com/davecgh/go-spew v1.1.1
go: project imports
        github.com/glebarez/sqlite imports
        github.com/glebarez/go-sqlite imports
        modernc.org/libc tested by
        modernc.org/libc.test imports
        modernc.org/ccgo/v4/lib: modernc.org/ccgo/v4@v4.10.1: verifying module: modernc.org/ccgo/v4@v4.10.1: Get "https://sum.golang.org/lookup/modernc.org/ccgo/v4@v4.10.1": dial tcp 142.251.43.17:443: connectex: A connection attempt failed because the connected party did not properly respond after a period of time, or established connection failed because connected host has failed to respond.
go: project imports
        github.com/glebarez/sqlite imports
        github.com/glebarez/go-sqlite imports
        modernc.org/libc tested by
        modernc.org/libc.test imports
        modernc.org/fileutil/ccgo: modernc.org/fileutil@v1.3.0: verifying module: modernc.org/fileutil@v1.3.0: Get "https://sum.golang.org/lookup/modernc.org/fileutil@v1.3.0": dial tcp 142.251.43.17:443: connectex: A connection attempt failed because the connected party did not properly respond after a period of time, or established connection failed because connected host has failed to respond.
```
Everything works fine when using goproxy.cn.
I deployed directly using the default Docker command without changing any configuration options.

EDIT: Perhaps I can set GOSUMDB to off to disable the verification to solve this problem, but my issue is that this problem does not exist when using goproxy.cn. I hope to keep my site consistent with it.

**Comment by aofei** [comment, marked as answer]
> but my issue is that this problem does not exist when using goproxy.cn

That's because goproxy.cn supports [proxying the default checksum database](https://golang.org/design/25530-sumdb#proxying-a-checksum-database) sum.golang.org.

You can set it up on your Goproxy instance if needed. See https://pkg.go.dev/github.com/goproxy/goproxy#Goproxy.ProxiedSumDBs.
